### ISSUE TITLE
make macro implementation method static

### ISSUE THREAD

**Issue by liufengyun** [liufengyun/gestalt#18]
As noted by @valdisxp1 , current implementation of macros doesn't support define macros inside inner classes.

With the following two PRs, we cleared the way to have static implementation of macros.

- https://github.com/lampepfl/dotty/pull/2057 (`merged`)
- https://github.com/lampepfl/dotty/pull/2058

**Comment by valdisxp1** [comment]
I am getting exceptions from the codegen when writing macros. For example, this does not compile:
```scala
inline def mapTest(): Int = meta {
  val sum = (1 to 5).map(_ * 2).sum
  toolbox.Lit(sum)
}
```
This means I cannot use anonymous functions or operations like `map`, `filter` etc.. It will likely be fixed by lampepfl/dotty#2058 The test case looks similar and I got a similar error as well.

**Comment by liufengyun** [comment]
I'm working on it, will release a new snapshot soon.
